source/compiler/qsc/src/packages.rs

```diff
@@ -75,19 +76,70 @@ fn convert_circuit_sources(
     processed_sources
 }
 
+#[must_use]
+pub fn get_target_profile_from_entry_point(
+    package_graph_sources: &PackageGraphSources,
+    dependency_errors: &mut Vec<WithSource<ErrorKind>>,
+) -> Option<Profile> {
+    // Convert circuit files in user code to generated Q# before entry profile check
+    let mut sources = package_graph_sources.root.sources.clone();
+    sources = convert_circuit_sources(sources, dependency_errors);
+
+    let converted_source_map = SourceMap::new(sources.clone(), None);
+
+    // Check if the entry profile is set in the source code.
+    let target_profile =
+        qsc_frontend::compile::get_target_profile_from_entry_point(&converted_source_map);
+
+    if let Some((profile, mut span)) = target_profile {
+        // If the entry profile is set, we need to ensure that the user code is compiled with it.
+        if package_graph_sources.has_manifest {
+            // Need to convert the span to the original source map
+            let original_source_map =
+                SourceMap::new(package_graph_sources.root.sources.clone(), None);
+            let converted_source = converted_source_map.find_by_offset(span.hi);
+            let original_source =
+                converted_source.and_then(|s| original_source_map.find_by_name(&s.name));
+            if let (Some(converted), Some(original)) = (converted_source, original_source) {
+                // Adjust the span to account for the offset of the original source
+                span = span - converted.offset + original.offset;
+            }
+
+            dependency_errors.push(Error::from_map(
+                &SourceMap::new(package_graph_sources.root.sources.clone(), None),
```

Comment:
I think you should be able to just pass in `converted_source_map` here and avoid the re-conversion?

Reply:
Yes, I was able to simplify the logic for handling circuit files by filtering them out of the parsing that looks for entry point target profiles.